client/app/account/billing/payment/ovhAccount/retrieve/billing-ovhAccount-retrieve.controller.js

```diff
@@ -1,69 +1,93 @@
-angular.module('Billing.controllers').controller('Billing.controllers.OvhAccountRetrieve', ($location, $q, $scope, $timeout, $translate, $window, Alerter, atInternet, BillingOvhAccount, BillingPaymentInformation, User, OVH_ACCOUNT_EVENT) => {
-  $scope.accountModel = $scope.currentActionData;
-  $scope.retrieve = {
-    amount: 0,
-    account: '',
-  };
+angular
+  .module('Billing.controllers')
+  .controller('Billing.controllers.OvhAccountRetrieve', (
+    $q,
+    $scope,
+    $state,
+    $timeout,
+    $translate,
+    $window,
+    Alerter,
+    atInternet,
+    BillingOvhAccount,
+    User,
+    OVH_ACCOUNT_EVENT,
+    ovhPaymentMethod,
+  ) => {
+    $scope.accountModel = $scope.currentActionData;
+    $scope.retrieve = {
+      amount: 0,
+      account: '',
+    };
 
-  $scope.initStep1 = () => {
-    $scope.loading = true;
-    $q
-      .all([
-        BillingPaymentInformation.getValidBankAccounts(),
-        User.getUser(),
-      ])
-      .then(([bankAccounts, user]) => {
-        $scope.bankAccounts = bankAccounts;
-        $scope.user = user;
-      })
-      .finally(() => {
-        $scope.loading = false;
-      });
-  };
+    function getValidBankAccounts() {
+      return ovhPaymentMethod.getAllPaymentMethods({
+        onlyValid: true,
+        transform: true,
+      }).then(paymentMethods => _.filter(paymentMethods, paymentMethod => paymentMethod.paymentType.value === 'BANK_ACCOUNT'));
+    }
+
+    $scope.initStep1 = () => {
+      $scope.loading = true;
+      $q
+        .all([
+          getValidBankAccounts(),
+          User.getUser(),
+        ])
+        .then(([bankAccounts, user]) => {
+          $scope.bankAccounts = bankAccounts;
+          $scope.user = user;
+        })
+        .finally(() => {
+          $scope.loading = false;
+        });
+    };
 
-  $scope.goToAddBankAccount = () => {
-    $scope.resetAction();
-    $timeout(() => $location.path('/billing/mean/add'), 300);
-  };
+    $scope.goToAddBankAccount = () => {
+      $scope.resetAction();
+      $timeout(() => $state.go('app.account.billing.payment.method.add', {
```

Comment:
Is the `$timeout` still needed ?

Reply:
Yes because without the `$timeout` the backdrop is not hidden :-(